debian/rules

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/make -f
+#export DH_VERBOSE = 1
+
+export GOCACHE := $(shell mktemp -d /tmp/gocache-XXXX)
+export GOFLAGS := -ldflags=-X=github.com/ubuntu/authd/internal/consts.Version=$(shell dpkg-parsechangelog -S Version) --mod=vendor -buildmode=pie
+
+export DEB_BUILD_MAINT_OPTIONS := optimize=-lto
+
+# Strict symbols checking
+export DPKG_GENSYMBOLS_CHECK_LEVEL := 4
+
+# Copy in build directory all content to embed
+export DH_GOLANG_INSTALL_ALL := 1
+
+# The following definitions are necessary because of the manual steps
+# we need to do to work around some issues with either dh-cargo,
+# the wrapper, or cargo
+include /usr/share/rustc/architecture.mk
+CARGO := /usr/share/cargo/bin/cargo
+
+# Needed for Rust vendored sources tracking
+DH_CARGO_VENDORED_SOURCES := /usr/share/cargo/bin/dh-cargo-vendored-sources
+export CARGO_VENDOR_DIR = vendor_rust
+
+# Needed for Rust configure/build/test stages
+export DEB_HOST_GNU_TYPE
+export DEB_HOST_RUST_TYPE
+export CARGO_HOME = $(CURDIR)/debian/cargo_home
+
+%:
+	dh $@ --buildsystem=golang --with=golang,apport
+
+override_dh_auto_clean:
+	dh_auto_clean
+	dh_auto_clean --buildsystem=cargo
+
+	# Create the vendor directory when building the source package
+	[ -d vendor/ ] || go mod vendor
+
+	# Some crates are shipped with .a files, which get removed by the helpers during the package build as a safety measure.
+	# This results in cargo failing to compile, since the files (which are listed in the checksums) are not there anymore.
+	# For those crates, we need to replace their checksum with a more general one that only lists the crate checksum, instead of each file.
+	if [ ! -d $(CARGO_VENDOR_DIR)/ ]; then \
+		CARGO_HOME=$(HOME)/.cargo $(CARGO) vendor $(CARGO_VENDOR_DIR); \
+		[ ! -e $(DH_CARGO_VENDORED_SOURCES) ] || $(DH_CARGO_VENDORED_SOURCES); \
+		\
+		[ -e /usr/bin/jq ] || (echo "jq is required to run this script. Try installing it with 'sudo apt install jq'" && exit 1); \
```

Comment:
Ouch, this is not in `debian/control` as build-deps, please add it.

Reply:
This shouldn't be in `debian/control`. It's only supposed to be executed once and locally (for the first build that we do before uploading the package). After the dependencies get vendored and the checksum gets updated, this is not supposed to be executed anymore.

Reply:
everything that is used in `debian/rules` has to be in `debian/control`, as build-dependency.

This is used to generate the package source, so it's definitely a build dependency.